rust/lance/src/io/exec/filtered_read.rs

```diff
@@ -520,18 +537,22 @@ impl FilteredReadStream {
     // We need to figure out which ranges to read from each fragment.
     //
     // If the scan range is ignoring the filters we can push it down here.
-    // If the scan range is not ignoring the filters we can only push it down if the index
-    // result is an exact match.
+    // If the scan range is not ignoring the filters we can only push it down if:
+    // 1. The index result is an exact match (we know exactly which rows will be in the result)
+    // 2. The index result is AtLeast with guaranteed rows >= limit (we have enough guaranteed matches)
+    // Returns: (fragment reads, whether limit was pushed down to fragment ranges)
     #[instrument(name = "plan_scan", skip_all)]
     async fn plan_scan(
         dataset: &Dataset,
         fragments: Vec<LoadedFragment>,
         evaluated_index: &Option<Arc<EvaluatedIndex>>,
-        options: FilteredReadOptions,
+        options: &FilteredReadOptions,
+        _global_metrics: &FilteredReadGlobalMetrics,
```

Comment:
why are we adding _global_metrics here if we don't use it?

Reply:
Thanks, removed the global metrics